//
// Academic License - for use in teaching, academic research, and meeting
// course requirements at degree granting institutions only.  Not for
// government, commercial, or other organizational use.
//
// validateattributes.h
//
// Code generation for function 'validateattributes'
//

#ifndef VALIDATEATTRIBUTES_H
#define VALIDATEATTRIBUTES_H

// Include files
#include "rtwtypes.h"
#include "coder_array.h"
#include <cstddef>
#include <cstdlib>

// Function Declarations
namespace coder {
void b_validateattributes(const ::coder::array<double, 2U> &a,
                          const double attributes_f2[2]);

void c_validateattributes(const ::coder::array<double, 2U> &a,
                          const double attributes_f2[2]);

void d_validateattributes(const ::coder::array<double, 2U> &a,
                          const double attributes_f2[2]);

void e_validateattributes(const ::coder::array<double, 2U> &a,
                          const double attributes_f2[2]);

void f_validateattributes(const ::coder::array<double, 2U> &a,
                          const double attributes_f2[2]);

void g_validateattributes(const ::coder::array<double, 2U> &a,
                          const double attributes_f2[2]);

void h_validateattributes(const ::coder::array<bool, 2U> &a,
                          const double attributes_f2[2]);

void i_validateattributes(const ::coder::array<bool, 2U> &a,
                          const double attributes_f2[2]);

void j_validateattributes(const ::coder::array<bool, 2U> &a,
                          const double attributes_f2[2]);

void k_validateattributes(const ::coder::array<bool, 2U> &a,
                          const double attributes_f2[2]);

void l_validateattributes(const ::coder::array<bool, 2U> &a,
                          const double attributes_f2[2]);

void m_validateattributes(const ::coder::array<bool, 2U> &a,
                          const double attributes_f2[2]);

void n_validateattributes(const ::coder::array<bool, 2U> &a,
                          const double attributes_f2[2]);

void validateattributes(double a);

void validateattributes(const ::coder::array<double, 2U> &a,
                        const double attributes_f2[2]);

void validateattributes(const ::coder::array<signed char, 2U> &a,
                        const double attributes_f2[2]);

void validateattributes(const ::coder::array<double, 1U> &a,
                        const double attributes_f2[2]);

void validateattributes(const ::coder::array<double, 3U> &a,
                        const double attributes_f2[3]);

} // namespace coder

#endif
// End of code generation (validateattributes.h)
